### PR TITLE
simplify code

### PR DESCRIPTION
**Pull request by rprtr258**
- change `interface{}` to `any`
- move `templateLen`, `formattedStr` after early exit, as they are not needed before it
- reduce in loop indent with `if ... { ... continue } else { ... }` to `if ... { ... continue} ...`
- simplify switch-case in `getItemAsStr` func
- prettify tests code, organize into `stringFormatter_test` package in order not to export some test symbols, organize tests into table tests
- change `MapToString` to accept just map, instead of pointer to it, and format instead of predefined strings, preallocate buffer which reduced memory consumption a bit
- updated readme with according changes and some fixes

### CODE REVIEW

**Comment by EvilLord666** [comment]
All tests are working, benchmark still shows slight performance advance in comparison to `fmt`, moving to code review

File: formatter_test.go [hunk 1]
```diff
@@ -1,145 +1,165 @@
...
-	strFormatResult = Format(strFormat4, v13, v14)
-	assert.Equal(t, "Here we are testing \"bool\" args: false, true", strFormatResult)
+const _address = "grpcs://127.0.0.1"
 
-	strFormat5 := "Here we are testing \"complex64\" {0} and \"complex128\": {1}"
-	var v15 complex64 = complex(1.0, 6.0)
-	var v16 complex128 = complex(2.3, 3.2)
-	strFormatResult = Format(strFormat5, v15, v16)
-	assert.Equal(t, "Here we are testing \"complex64\" (1+6i) and \"complex128\": (2.3+3.2i)", strFormatResult)
+type Example struct {
```
Comment (EvilLord666) [inline formatter_test.go:14]: I suggest to rename `type Example struct` to `type meteoData struct`

**Review by EvilLord666 — CHANGES_REQUESTED** [review]
Minor changes are required: `Example` should be renamed to more clear type i.e. meteoData and should be made visible only for _test package

File: formatter_test.go (hunk 1 above)
Comment (EvilLord666) [reply formatter_test.go:14]: @rprtr258, checked you changes, great job, but there is a minor code review issue

Comment (rprtr258) [reply formatter_test.go]: renamed to `meteoData`, also, as i know, nothing from `*_test` packages is exported and doesn't even have to compile, so no worry, to leak that struct even if it was exported

**Pull request merged**